### PR TITLE
✨ Enhancements to Davis Interactive for 2019 version.

### PR DESCRIPTION
**Pull request by albertomontesg**
TODO(scaelles): write documentation.

### CODE REVIEW

**Comment by scaelles** [comment]
Would it be possible to change 'next_scribble_frame: Integer' to 'next_scribble_frames: List' where instead of an integer the user provides a range of possible frames and we return the scribbles for the worst prediction in that subset of frames?

**Comment by albertomontesg** [comment]
> Would it be possible to change 'next_scribble_frame: Integer' to 'next_scribble_frames: List' where instead of an integer the user provides a range of possible frames and we return the scribbles for the worst prediction in that subset of frames?

I think this can be a point of discussion. If nothing is given the worst frame is chosen and its scribble return, and if one frame is given the scribble for that frame is given. Accepting a list will be that half the logic will be done by us (choosing the worst frame) and then the other half done by the user. It seems like a equally distant approach where I can not see the real scenario applications.

File: davisinteractive/connector/abstract.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def get_scribble(self, sequence, scribble_idx):
         raise NotImplementedError('This is an abstract class')
 
     def post_predicted_masks(self, sequence, scribble_idx, pred_masks, timming,
-                             interaction):
+                             interaction, next_scrrible_frame_candidates=None):
```
Comment (scaelles) [inline davisinteractive/connector/abstract.py]: It should be in next_scribble_frame_candidates

Comment (scaelles) [inline davisinteractive/connector/abstract.py]: It should be next_scribble_frame_candidates

Comment (albertomontesg) [reply davisinteractive/connector/abstract.py]: Done

File: davisinteractive/evaluation/service.py [hunk 2]
```diff
@@ -166,11 +178,15 @@ def post_predicted_masks(self, sequence, scribble_idx, pred_masks, timing,
         gt_masks = self.davis.load_annotations(sequence)
         nb_objects = Davis.dataset[sequence]['num_objects']
 
-        jaccard = batched_jaccard(
+        jaccard = 0.5 * batched_jaccard(
```
[the remark this answers is not in the document]
Comment (albertomontesg) [reply davisinteractive/evaluation/service.py]: Done

File: davisinteractive/evaluation/service.py [hunk 3]
```diff
@@ -140,6 +157,11 @@ def post_predicted_masks(self, sequence, scribble_idx, pred_masks, timing,
             interaction: Integer. Interaction number.
             user_key: String. User identifier.
             session_key: String. Session identifier.
+            next_scribble_frame_candidates: List of Integers. Optional value
+                specifying the possible frames from which generate the next
+                scribble. If values given, the next scribble will be performed
+                in the frame where the evaluation metric score the least on the
```
Comment (scaelles) [inline davisinteractive/evaluation/service.py]: scores

File: davisinteractive/session/session.py [hunk 4]
```diff
@@ -259,14 +263,19 @@ def scribbles_iterator(self, *args, **kwargs):
...
 
         # Arguments
             pred_masks: Numpy array with the predicted mask for
                 the current sample. The array must be of `dtype=np.int` and
                 of size equal to the 480p resolution of the DAVIS
                 dataset.
+            next_scribble_frame_candidates: List of Integers. Optional value
+                specifying the possible frames from which generate the next
+                scribble. If values given, the next scribble will be performed
+                in the frame where the evaluation metric score the least on the
```
Comment (scaelles) [inline davisinteractive/session/session.py]: scores

**Pull request merged**